src/lib/bootloader/systemdboot_widgets.rb

```diff
@@ -17,35 +17,63 @@ def systemdboot
     end
 
     # Represents bootloader timeout value
-    class TimeoutWidget < CWM::IntField
+    class TimeoutWidget < CWM::CustomWidget
       include SystemdBootHelper
 
       def initialize
         textdomain "bootloader"
 
         super()
 
-        @minimum = -1
+        @minimum = 0
         @maximum = 600
+        @default = 10
       end
 
-      attr_reader :minimum, :maximum
+      attr_reader :minimum, :maximum, :default
 
-      def label
-        _("&Timeout in Seconds")
+      def contents
+        CheckBoxFrame(
+          Id(:cont_boot),
+          _("Automatically boot the default entry after a timeout"),
+          false,
+          HBox(
+            IntField(Id(:seconds), _("&Timeout in Seconds"), @minimum, @maximum,
+              systemdboot.menue_timeout.to_i),
+            HStretch()
+          )
+        )
       end
 
       def help
-        _("<p><b>Timeout in Seconds</b>\n" \
+        _("<p>Continue boot process after defined seconds.</p>" \
+          "<p><b>Timeout in Seconds</b>\n" \
           "specifies the time the boot loader will wait until the default kernel is loaded.</p>\n")
       end
 
       def init
-        self.value = systemdboot.menue_timeout.to_i
+        Yast::UI.ChangeWidget(Id(:cont_boot), :Value, systemdboot.menue_timeout >= 0)
+        systemdboot.menue_timeout = default_value if systemdboot.menue_timeout < 0
+        Yast::UI.ChangeWidget(Id(:seconds), :Value, systemdboot.menue_timeout)
       end
 
       def store
-        systemdboot.menue_timeout = value.to_s
+        if Yast::UI.QueryWidget(Id(:cont_boot), :Value)
+          systemdboot.menue_timeout = Yast::UI.QueryWidget(Id(:seconds), :Value)
+          systemdboot.menue_timeout = default_value if systemdboot.menue_timeout == -1
```

Comment:
Can this ever be -1 here, given that it's read from the widget here?

Reply:
Yes it could if Yast::UI.QueryWidget(Id(:cont_boot), :Value) is set. menue_timeout is set to the default value in orde r not to see -1 for the user anymore. I have tested it. 
You have requested it ;-)

Reply:
`Yast::UI.QueryWidget(Id(:seconds), :Value)` is controlled by yast though and has a minimum of 0

Reply:
Not resolved yet, but apparently I can't reopen this thread